tiledb/sm/query/sparse_index_reader_base.h

```diff
@@ -65,6 +65,46 @@ class ResultTileWithBitmap : public ResultTile {
       , qc_processed(false) {
   }
 
+  /* ********************************* */
+  /*          PUBLIC METHODS           */
+  /* ********************************* */
+
+  /**
+   * Returns the number of cells that are before a certain cell index in the
+   * bitmap.
+   */
+  uint64_t num_cells_between_idexes(uint64_t start, uint64_t end) const {
+    if (bitmap.size() == 0)
+      return end - start;
+
+    uint64_t sum = 0;
+    for (uint64_t c = start; c < end; c++)
+      sum += bitmap[c];
+
+    return sum;
+  }
+
+  /**
+   * Returns cell index from a number of cells inside of the bitmap.
```

Comment:
You mean that this returns the bitmap position at and before which we hit the same the number of results (1s) specified in the input. Therefore, let's call the function `pos_with_given_result_sum` and the input to `result_sum` or something along those lines.